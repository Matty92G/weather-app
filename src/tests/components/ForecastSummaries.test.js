import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 11111111,
        icon: "stubIcon1",
        description: "stubDescription1",
        temperature: {
          min: 10,
          max: 91,
        },
      },
      {
        date: 22222222,
        icon: "stubIcon2",
        description: "stubDescription2",
        temperature: {
          min: 20,
          max: 92,
        },
      },
    ],
    onForecastSelect: () => {},
  };
  it("renders and matches snapshot", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps.forecasts} />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
