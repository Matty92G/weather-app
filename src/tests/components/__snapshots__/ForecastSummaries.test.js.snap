// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <img
          alt="NaN"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        91°C
      </div>
      <div
        class="forecast-summary_description"
      >
        stubDescription1
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <img
          alt="NaN"
        />
      </div>
      <div
        class="forecast-summary_temperature"
      >
        92°C
      </div>
      <div
        class="forecast-summary_description"
      >
        stubDescription2
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
